### PR TITLE
Feature/weatherapp

### PR DESCRIPTION
**Pull request by NewIncome**
This branch creates a weather app with the following functionalities:
- searches the weather by city name
- gives an error if city not found
- lets you swap from °C to °F
- shows an image of the flag of the country
- displays the current temperate, temperature feel, max and min temperatures
- also displays the date, the wind speed and the humidity
- it displays a picture of the city if found or else a random city image
- lastly it changes the color of the background depending on the temperature (only 4 ranges)

### CODE REVIEW

File: src/javascript/dom-work.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+const makeNewDate = () => {
+  const Months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+  const date = new Date();
+  return `${Months[date.getMonth() + 1]} ${date.getDate()}, ${date.getFullYear()}`;
+};
+
+const updateInfo = (values, city) => {
```
Comment (wathigo) [inline src/javascript/dom-work.js]: We can use objtect destructuring to get all the items in values object. For example, we can get
```js
const { TempDT, countryDt } = values;
```
We can then access `TempDT` and `countryDt` directly without using repeating `values `.
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/destructuring_requirement.png) from Pathwright).

File: package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "build": "webpack",
-    "watch": "webpack --watch"
+    "watch": "webpack --watch",
+    "server": "webpack-dev-server"
```
Comment (wathigo) [inline package.json]: **OPTIONAL**
Let's change `"server": "webpack-dev-server"` to `"start": "webpack-dev-server"` to get rid of an error raised on an attempt to start the server.

File: src/javascript/api-data.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+async function getTemp(newCity = 'Toronto') {
+  const weatherData = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${newCity}&units=metric&appid=550a45bc4aa9613e4687926216e69765`)
+    .then((response) => response.json()).then((data) => data);
```
Comment (wathigo) [inline src/javascript/api-data.js]: **REQUIRED**
Let's add error handling to handle the case when an error is thrown while processing requests. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

File: src/javascript/app-controller.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {
+  render,
+  checkForClick,
+  changeTemp,
+} from './dom-work';
+
+function start() {
```
Comment (wathigo) [inline src/javascript/app-controller.js]: **REQUIRED**
We should strictly use ES6 syntax. Let's replace function with syntax with and arrow function. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) from Pathwright)

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required.
Great work guys :clap: 
Let's make a couple of changes before we move on.
Check the comments :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :heavy_check_mark: 
Simon Wathigo**

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request closed without merging**